frontend/src/pages/dashboard/beta/gym/index.tsx

```diff
@@ -51,38 +65,141 @@ const dot = (color = 'transparent') => ({
   },
 });
 
-const FAKE_BETA: {
-  author: Partial<User>;
-  beta: BetaDemo;
-} = {
-  author: {
-    telegramHandle: '@rizhaow',
-  },
-  beta: {
-    imageUrl: 'https://i.ibb.co/k1yHPSz/image-27.png',
-    color: 'Red',
-    grade: '3 Bar',
-    wall: 'Slab',
-    gym: 'FitBloc',
-    createdAt: new Date(),
+const StyledInfiniteScroll = styled(InfiniteScroll)({
+  maxWidth: 600,
+  width: '100vw',
+  display: 'grid',
+  gridTemplateColumns: '1fr 1fr',
+  columnGap: 16,
+  rowGap: 16,
+  padding: '0 12px',
+  '& > div:first-of-type': {
+    gridColumn: 'span 2',
   },
-};
+});
+
+const InfiniteScrollHelper = styled((props: TypographyProps) => (
+  <Typography {...props} variant="h5" />
+))({
+  gridColumn: 'span 2',
+  textAlign: 'center',
+  justifySelf: 'center',
+});
+
+const ALL_VALUE = undefined;
+
+const addAllOption = (list: { value: number; label: string }[]) => [
+  // By default, if the value is undefined, we will fetch all data
+  { value: ALL_VALUE, label: 'All' },
+  ...list,
+];
 
 export default function BetaGym() {
+  // Number of Betas to fetch per page
+  const PAGE_SIZE = 10;
+  const [selectedGymGrade, setSelectedGymGrade] = useState<GymGrade['id'] | undefined>(undefined);
+  const [selectedWall, setSelectedWall] = useState<Wall['id'] | undefined>(undefined);
+  const [selectedColor, setSelectedColor] = useState<Color['id'] | undefined>(undefined);
+  const errorSnackbar = useErrorSnackbar();
+  // True iff user is scrolling
   const trigger = useScrollTrigger({
     target: window,
   });
-  // TODO: Fetch and set data
-  const colorOptions = COLORS;
-  const wallOptions = WALLS;
-  const gradeOptions = GRADES;
-
   const theme = useTheme();
-  const { gymId } = useParams();
-  const gym = useSelector((state) => state.gyms.data.find((gym) => gym.id === Number(gymId)));
+  const params = useParams();
+  const gymId = Number(params.gymId);
+  const gym = useSelector((state) => state.gyms.data.find((gym) => gym.id === gymId));
```

Comment:
shall we use useRequest for gyms instead of redux ?

Reply:
Yes